File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    float l = 0, w = 1, s = 0;
    string input = get_string("Text: ");
    int len = strlen(input);
    for (int i = 0; i < len; i++)
    {
        char x = input[i];
        char prevx = input[i-1];
        if(isalpha(x))
        {
            l++;
        }
        else if (isspace(x) && !isspace(prevx))
        {
            w++;
        }
        else if(x == '.' || x == '!' || x == '?')
        {
            s++;
        }
    }
    l *= (100 / w);
    s *= (100 / w);
    float output = round(0.0588 * l - 0.296 * s - 15.8);
    if(output < 1) 
    {
        printf("Before Grade 1\n");
    }
    else if(output > 16)
    {
        printf("Grade 16+\n");
    }
    else 
    {
        printf("Grade %i\n", (int) output);
    }
}

File: readability.py
l = float(0)
w = float(1)
s = float(0)
txt = input("Text: ")
for i in range(len(txt)):
    x = txt[i]
    prevx = txt[i - 1]
    if x.isalpha():
        l += 1
    elif x.isspace() and not prevx.isspace():
        w += 1
    elif x in [".", "!", "?"]:
        s += 1
l *= (100 / w)
s *= (100 / w)
output = round(0.0588 * l - 0.296 * s - 15.8)
if output < 1:
    print("Before Grade 1")
elif output > 16:
    print("Grade 16+")
else:
    print("Grade", output)